library/cgi/escapeURIComponent_spec.rb

```diff
@@ -0,0 +1,41 @@
+require_relative '../../spec_helper'
+require 'cgi'
+
+describe "CGI.escapeURIComponent" do
+  ruby_version_is "3.2" do
+    it "escapes whitespace" do
+      str = "&<>\" \xE3\x82\x86\xE3\x82\x93\xE3\x82\x86\xE3\x82\x93"
+      CGI.escapeURIComponent(str).should == '%26%3C%3E%22%20%E3%82%86%E3%82%93%E3%82%86%E3%82%93'
+    end
+
+    it "does not escape with unreserved characters" do
+      str = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~"
+      CGI.escapeURIComponent(str).should == "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~"
+    end
+
+    it "with invalid bytesequence" do
```

Comment:
**suggestion:** a test title should usually be a regular sentence e.g.:
```suggestion
    it "supports String with invalid encoding" do
```

or just `"does not raise error if String encoding is invalid"`

Reply:
Could you please also review titles of all the other cases, e.g. 

- `it "empty string" do`
- `it "single whitespace" do`
- `it "double whitespace" do`